src/opentracing/Tracer.ts

```diff
@@ -0,0 +1,143 @@
+import { Tracer, Reference, Tags } from 'opentracing';
+import * as opentracing from 'opentracing';
+import ThundraSpan, { SpanEvent } from './Span';
+import ThundraSpanContext from './SpanContext';
+import ThundraRecorder from './Recorder';
+import ThundraSampler from './Sampler';
+import Utils from '../plugins/Utils';
+
+class ThundraTracer extends Tracer {
+  tags: any;
+  recorder: ThundraRecorder;
+  sampler: ThundraSampler;
+  activeSpans: Map<string, ThundraSpan>;
+
+  constructor(config: any = {}) {
+    super();
+
+    this.tags = config.tags;
+    this.recorder = config.recorder ? config.recorder : new ThundraRecorder();
+    this.sampler = config.sampler ? config.sampler : new ThundraSampler(1);
+    this.activeSpans = new Map<string, ThundraSpan>();
+  }
+
+  getActiveSpan(): ThundraSpan {
+    return this.recorder.getActiveSpan() ? this.recorder.getActiveSpan().value : null;
+  }
+
+  finishSpan(): void {
+    if (this.getActiveSpan()) {
+       this.getActiveSpan().finish();
+    }
+  }
+
+  getRecorder(): ThundraRecorder {
+    return this.recorder;
+  }
+
+  destroy() {
+    this.recorder.destroy();
+    this.activeSpans = null;
+  }
+
+  wrapper<T extends (...args: any[]) => any>(spanName: string, func: T): T {
+    // tslint:disable-next-line:no-angle-bracket-type-assertion
+    return <T> ((...args: any[]) => {
```

Comment:
I think, we should get active span and store it to a local variable here. Then while starting new span, the new span is attached to that stored span instead of active span while running callback. 

So with this way, callbacks can be attached to the spans which are active while creating callbacks.

Reply:
👍